lib/ui/window.dart

```diff
@@ -425,6 +425,16 @@ class SingletonFlutterWindow extends FlutterWindow {
   ///    observe when this value changes.
   double get textScaleFactor => platformDispatcher.textScaleFactor;
 
+  /// The setting inidicating whether or not Flutter currently supports
+  /// spell checking via the native spell check service.
+  ///
+  /// {@macro dart.ui.window.accessorForwardWarning}
+  ///
+  /// This option is used by [EditableTextState] to define its
+  /// [SpellCheckConfiguration] when spell check is enabled, but no spell check
+  /// service is specified.
+  bool get nativeSpellCheckServiceDefined => platformDispatcher.nativeSpellCheckServiceDefined;
```

Comment:
what is the difference between this and the properties in lib/ui/platform_dispatcher.dart?

Reply:
Can this be part of the framework and the logic can be something like:
```
nativeSpellCheckServiceDefined = Platform == Android || Platform == iOS
```

Reply:
I thought the benefit of getting this information from the engine is that we can check more detailed stuff than just the platform, like the API version of Android, or even if the device has some third party spell checker installed? If that's a thing. But if that stuff doesn't matter and we can decide based on the platform alone, then totally this should be in the framework +1

Reply:
We are able to get some information to inform whether or not spell check is supported on a particular device on API 31+, so I kept it in the engine!